src/pmtiles_adapter.js

```diff
@@ -0,0 +1,111 @@
+import fs from 'node:fs';
+import PMTiles from 'pmtiles';
+
+const PMTilesLocalSource = class {
+  constructor(file) {
+    this.file = file;
+  }
+  getKey() {
+    return this.file.name;
+  }
+  async getBytes(offset, length) {
+    const buffer = Buffer.alloc(length);
+    const fd = fs.openSync(this.file, 'r'); //Open the file in read mode
+    await ReadBytes(fd, buffer, offset); //Read the specifed bytes from the file
+    fs.closeSync(fd); //close the file
+    return { data: BufferToArrayBuffer(buffer) };
+  }
```

Comment:
I added PMtilesOpen, which opens a file descriptor to the file and I changed all the pmtiles functions to expect a file descriptor instead of a filename.

I also added PMtilesClose to close, but i'm not 100% sure where it needs to go yet.

Reply:
I reworked it a second time so now PMtilesOpen passes back a pmtiles object, which get passes to the other functions.

I think doing this might make it possible to support remote pmtiles too, but that will need some work and testing. My thinking is if you wanted remote pmtiles, you would specify a url into PMtilesOpen  instead of a file path.